### PR TITLE
Release 2.4.0

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: History.md [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+* **fix**
+  * add logger in base_context_class
+
+* **docs**
+  * Lots of d.ts improved.
+    * add context
+    * add urllib
+    * add resources & logger
+  * new documents
+    * how to call the service
```
Comment (atian25) [inline History.md]: 怎么印象中还有个 alinode 文档的，是之前发了么？

Comment (atian25) [reply History.md]: 果然上个版本发了，这个版本这几个文档应该值得注意

- https://github.com/eggjs/egg/pull/2098
- https://github.com/eggjs/egg/pull/2039

File: History.md [hunk 2]
```diff
@@ -1,5 +1,51 @@
 # History
 
+## 2018-02-28, Version 2.4.0, @dead-horse
```
Comment (popomore) [inline History.md]: 名字没改？

**Comment by fengmk2** [comment]
都修改了

**Pull request merged**